src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -888,9 +912,19 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			this._activeConnection.id !== newConnection.id) {
 			// Change the active connection to newConnection
 			this._activeConnection = newConnection;
+			this._savedConnectionName = newConnection.connectionName;
 		}
 	}
 
+	private getConnectionProfileFromName(connectionName: string): ConnectionProfile | undefined {
+		let connections: ConnectionProfile[] = this.connectionManagementService.getConnections();
+		return values(connections).find(connection => connection.connectionName === connectionName);
```

Comment:
names are not guaranteed to be unique - are we sure we want to just grab the first one we find?